[ONNX] Fix exported onnx initializer name

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #101029
* __->__ #104741
* #104493

Restore ONNX initializer name to exactly match torch parameter and buffer name.
Fixes #104670 that could lead to potentially duplicated ONNX initializers after export.

BowenBao committed:
[ONNX] Fix initializer export

[ghstack-poisoned]

BowenBao commented on torch/onnx/_internal/fx/fx_onnx_interpreter.py:
```diff
@@ -657,23 +657,18 @@ def get_attr(
 
-        input_ = onnxscript_graph.add_initializer(node.name, current_attr)
+        input_ = onnxscript_graph.add_initializer(
+            node.target.replace("/", "."), attr_tensor
```
Depending on how much people hate this `replace` pair. I can leave this as is or make a much more complicated PR storing the name mapping into a custom meta class inside node.meta, and try retrieve the name from there.

thiagocrepaldi replied:
lgtm

BowenBao committed:
add a simple comment on "[ONNX] Fix exported onnx initializer name"


Restore ONNX initializer name to exactly match torch parameter and buffer name.
Fixed a bug that could lead to potentially duplicated ONNX initializers after export.

[ghstack-poisoned]

BowenBao commented on torch/onnx/_internal/fx/fx_onnx_interpreter.py:
```diff
@@ -657,23 +657,20 @@ def get_attr(
         ],
         fx_graph_module: torch.fx.GraphModule,
     ):
-        current_attr = fx_graph_module
```
I'm removing this parameter lookup from submodules because this is dead code. We do not get fx.GraphModule like this in our current code path.

thiagocrepaldi commented on torch/onnx/_internal/fx/fx_onnx_interpreter.py:
```diff
@@ -657,23 +657,20 @@ def get_attr(
+            raise TypeError(
+                f"node.target must be a str, not {type(node.target)}. Found for node {node}."
+            )
+        assert isinstance(node.target, str), f"node.target {node.target} is not a str."
```
the check on previous line `if not isinstance(node.target, str):` already handles that case

BowenBao replied:
Huh I meant to delete the previous block but forgot to do it. Since we are here, which version of the two do you prefer?

thiagocrepaldi replied:
no strong preference

BowenBao committed:
Update on "[ONNX] Fix exported onnx initializer name"


Restore ONNX initializer name to exactly match torch parameter and buffer name.
Fixes #104670 that could lead to potentially duplicated ONNX initializers after export.

[ghstack-poisoned]

BowenBao committed:
remove dup check on "[ONNX] Fix exported onnx initializer name"


Restore ONNX initializer name to exactly match torch parameter and buffer name.
Fixes #104670 that could lead to potentially duplicated ONNX initializers after export.

[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge
